### DISCUSSION TITLE
Why doesn't Vercel support 302 redirect caching?

### DISCUSSION THREAD

**Discussion by bashbaugh** [vercel/community#177] [Help] [answered]
Vercel supports [caching permanent redirects but not temporary redirects](https://vercel.com/docs/edge-network/caching#cacheable-responses). I have a use case (for a dynamic link shortener/redirector that's part of my website) where I want redirects to be cached so that they are served quickly without running my slow serverless function first every time. However, some of these redirects will change quite often, so permanent redirects are not viable. The `stale-while-revalidate` caching option would be perfect for my use case.

So, is there any reason 301 redirect caching (arguably much less useful) is allowed but not 302 redirect caching?

**Comment by wiiiimm** [comment, marked as answer]
1. `302` and `307` redirects are supposed to be temporary redirects meaning their effects should be temporary.
2. Using `stale-while-revalidate`, a stale cache can be served taking visitors to a staled location, which makes its effects less "temporary".

**Reply by bashbaugh** [reply]
That makes sense, however, intentionally serving stale redirects (when it fits your use case) doesn't seem much different than serving stale data of any other type, so I don't completely understand why it's not supported when I'm sure it would be trivial to support

**Reply by lucasconstantino** [reply]
@bashbaugh is entirely correct.

Many times, a dynamic redirect operation might be heavy, though its result might change not often.

On a CDN, we could have a 307 cached, while ensuring it gets invalidated when need – either TTL, or via active invalidation. That is the way to solve the problem, as for 308 we would not have control over the invalidation of ISPs.

Please, reconsider this.
